runtime/grid_sync.cu

```diff
@@ -150,4 +150,82 @@ __device__ void sync(
   block_sync::sync<Aligned>();
 }
 
+// Non-blocking function to acquire the semaphore value in each calling thread
+__device__ int64_t semaphoreFetch(int64_t* semaphore) {
+  int64_t state;
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 700
+  asm volatile("ld.global.acquire.gpu.b64 %0, [%1];\n"
+               : "=l"(state)
+               : "l"(semaphore));
+#else
+  asm volatile("ld.global.cg.b64 %0, [%1];\n" : "=l"(state) : "l"(semaphore));
+#endif
+  return state;
+}
+
+// Sync block then set semaphore to new_value
```

Comment:
Sync is not here, right?

Reply:
You're right. I need to update these comments.